js/src/hooks/useAdsCampaigns.js

```diff
@@ -30,19 +31,22 @@ const selectorName = 'getAdsCampaigns';
  */
 const useAdsCampaigns = ( ...query ) => {
 	const queryRefValue = useIsEqualRefValue( query );
+	const {
+		googleAdsAccount,
+		hasGoogleAdsConnection,
+		hasFinishedResolution,
+		isResolving,
+	} = useGoogleAdsAccount();
 
 	return useSelect(
 		( select ) => {
-			// TODO: ideally adsSetupComplete should be retrieved from API endpoint
-			// and then put into wp-data.
-			// With that in place, then we don't need to depend on glaData
-			// which requires force reload using window.location.href.
-			const { adsSetupComplete } = glaData;
-
-			if ( ! adsSetupComplete ) {
+			if (
+				! hasGoogleAdsConnection ||
```

Comment:
`hasGoogleAdsConnection` can be true for incomplete ads accounts(unclaimed). But we can't create or fetch campaign from those accounts, hence the next check.